### PR TITLE
Moves pledge calculator functionality from patients.coffee to pledg…

### LINKED ISSUE #1683: JS: Port pledge calculator part of patients.coffee from coffeescript to ES6
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

We're porting our custom javascript to ES6! This ticket is to port over:

patients.coffee into app/javascript/src/pledge_calculator.js

Note that this file contains a lot of unrelated functionality, so no need to port it all over at once!

* **What feature or behavior is this required for?**

Long term project health and maintainability, plus an opportunity to clean up some of our messes when we wrote this the first time.

* **How could we solve this issue? (Not knowing is okay!)**

Rewrite the coffeescript in app/assets/ to be ES6 in app/javascript/src.

* **Anything else?**

See #1679 for an example of this with the `clinic_finders` js.

If this winds up requiring adding dependencies or something, please hit me up in slack!

### PR DESCRIPTION
**Pull request by bravogolfgolf**
This pull request makes the following changes:
* Moves pledge calculator functionality to separate file

It relates to the following issue #s: 
* Fixes #1683

### CODE REVIEW

File: app/javascript/src/pledge_calculator.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    if ($("#patient_procedure_cost").val()) {
+        $(".outstanding-balance-ctn").removeClass('hidden');
+        return $("#outstanding-balance").text('$' + calculateRemainder());
+    } else {
+        return $(".outstanding-balance-ctn").addClass('hidden');
+    }
+};
+
+var calculateRemainder = function() {
+    var contributions, total;
```
Comment (colinxfleming) [inline app/javascript/src/pledge_calculator.js]: Here's another place where we can use javascript's `let` and `const` to tell other programmers that something's going to stay the same after we declare it, or that we're going to manipulate it. In this case, we're setting `total`, setting `contributions` by doing some fancy math, and then returning the difference. So we aren't really changing these once we're setting them -- what do you think about removing this line, and then doing `total = valueToNumber($("#patient_procedure_cost").val());` and `const contributions = [BUNCH OF FANCY MATH]`? In your opinion, does that make sense?

**Comment by bravogolfgolf** [comment]
I took your suggestions and incorporated them in two self-explanatory commits.

File: app/assets/javascripts/patients.coffee [hunk 2]
```diff
@@ -1,54 +1,6 @@
...
-                  valueToNumber($("#patient_fund_pledge").val()) +
-                  valueToNumber($(".external_pledge_amount").toArray().reduce (acc, next) ->
-                    acc + valueToNumber($(next).val())
-                  , 0)
-  total - contributions
-
-valueToNumber = (val) ->
-  +val || 0
-
-markFulfilledWhenFieldsChecked = ->
```
[the remark this answers is not in the document]
Comment (bravogolfgolf) [reply app/assets/javascripts/patients.coffee]: Restored markFulfilledWhenFieldsChecked() to patients.coffee

File: app/assets/javascripts/patients.coffee [hunk 3]
```diff
@@ -58,10 +10,6 @@ $(document).on 'turbolinks:load', ->
   $(document).on "change", ".edit_patient", ->
     $(this).submit()
 
-  $(document).on "change", "#pledge_fulfillment_form", ->
```
[the remark this answers is not in the document]
Comment (bravogolfgolf) [reply app/assets/javascripts/patients.coffee]: Restored to patients.coffee

File: app/assets/javascripts/patients.coffee [hunk 4]
```diff
@@ -75,13 +23,6 @@ $(document).on 'turbolinks:load', ->
   $(document).on "change", balanceFields, ->
     updateBalance()
 
-  $(document).on "submit", "#generate-pledge-form form", ->
```
[the remark this answers is not in the document]
Comment (bravogolfgolf) [reply app/assets/javascripts/patients.coffee]: Restored to patients.coffee

File: app/javascript/src/pledge_calculator.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        return acc + valueToNumber($(next).val());
+    }, 0));
+    return total - contributions;
+};
+
+var valueToNumber = function(val) {
+    return +val || 0;
+};
+
+var markFulfilledWhenFieldsChecked = function() {
```
[the remark this answers is not in the document]
Comment (bravogolfgolf) [reply app/javascript/src/pledge_calculator.js]: Restored markFulfilledWhenFieldsChecked() to patients.coffee

File: app/javascript/src/pledge_calculator.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+var updateBalance = function() {
```
[the remark this answers is not in the document]
Comment (bravogolfgolf) [reply app/javascript/src/pledge_calculator.js]: Updated pledge_calculator.js to take advantage of ES6 feature

**Comment by colinxfleming** [comment]
thank you for sticking on this @bravogolfgolf !

**Pull request merged**